client/components/backup-storage-space/usage-warning/use-upsell-slug.ts

```diff
@@ -0,0 +1,89 @@
+import {
+	PRODUCT_JETPACK_BACKUP_T1_MONTHLY,
+	PRODUCT_JETPACK_BACKUP_T1_YEARLY,
+	PLAN_JETPACK_SECURITY_T1_MONTHLY,
+	PLAN_JETPACK_SECURITY_T1_YEARLY,
+	PRODUCT_JETPACK_BACKUP_ADDON_10GB_MONTHLY,
+	PRODUCT_JETPACK_BACKUP_ADDON_1TB_MONTHLY,
+	useJetpack100GbStorageAmountText,
+	useJetpack10GbStorageAmountText,
+	useJetpack1TbStorageAmountText,
+	PRODUCT_JETPACK_BACKUP_T2_MONTHLY,
+	PRODUCT_JETPACK_BACKUP_T2_YEARLY,
+	PLAN_JETPACK_SECURITY_T2_MONTHLY,
+	PLAN_JETPACK_SECURITY_T2_YEARLY,
+} from '@automattic/calypso-products';
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import slugToSelectorProduct from 'calypso/my-sites/plans/jetpack-plans/slug-to-selector-product';
+import { SelectorProduct } from 'calypso/my-sites/plans/jetpack-plans/types';
+import useItemPrice from 'calypso/my-sites/plans/jetpack-plans/use-item-price';
+import { getSitePurchases } from 'calypso/state/purchases/selectors';
+
+type SelectorProductWithStorage = SelectorProduct & {
+	storage: React.ReactChild;
+};
+const UPGRADEABLE_STORAGE_PRODUCT_SLUGS_T1 = [
+	PRODUCT_JETPACK_BACKUP_T1_MONTHLY,
+	PRODUCT_JETPACK_BACKUP_T1_YEARLY,
+	PLAN_JETPACK_SECURITY_T1_MONTHLY,
+	PLAN_JETPACK_SECURITY_T1_YEARLY,
+];
+
+const UPGRADEABLE_STORAGE_PRODUCT_SLUGS_T2 = [
+	PRODUCT_JETPACK_BACKUP_T2_MONTHLY,
+	PRODUCT_JETPACK_BACKUP_T2_YEARLY,
+	PLAN_JETPACK_SECURITY_T2_MONTHLY,
+	PLAN_JETPACK_SECURITY_T2_YEARLY,
+];
+
+export const isJetpackT1Product = ( { subscriptionStatus, productSlug } ) => {
+	return (
+		subscriptionStatus === 'active' && UPGRADEABLE_STORAGE_PRODUCT_SLUGS_T1.includes( productSlug )
+	);
+};
+
+export const isJetpackT2Product = ( { subscriptionStatus, productSlug } ) => {
+	return (
+		subscriptionStatus === 'active' && UPGRADEABLE_STORAGE_PRODUCT_SLUGS_T2.includes( productSlug )
+	);
+};
+
+export default ( siteId ) => {
+	const sitePurchases = useSelector( ( state ) => getSitePurchases( state, siteId ) );
+	const TEN_GIGABYTES = useJetpack10GbStorageAmountText();
+	//const HUNDRED_GIGABYTES = useJetpack100GbStorageAmountText();
+	const ONE_TERABYTES = useJetpack1TbStorageAmountText();
+
+	const currencyCode = sitePurchases?.[ 0 ]?.currencyCode;
```

Comment:
I am not exactly sure how to get the current user currency, I am currently using the same currency they used on their previous purchase. Any help on this is greatly appreciated.

Reply:
For logged-in sessions, this selector should get you the current user's currency code:

```js
import { getCurrentUserCurrencyCode} from 'calypso/state/currency-code/selectors';
const currencyCode = useSelector( getCurrentUserCurrencyCode );
```

Reply:
I did try this, but the `currencyCode` in the state was actually `null` even for the logged-in users. However, for the same user in the pricing page, I am able to see the currency set

Reply:
Hmm.. 🤔 I can't figure out why the `getCurrentUserCurrencyCode` selector is always returning null. 🤔🤔

Reply:
Ahh.. I think see now why `getCurrentUserCurrencyCode` isn't working..  The `currencyCode` state value isn't populated or updated unless/until `<QueryProductsList/>` or `<QueryPlans />` is executed..  
So maybe add `<QueryProductsList />' to `client/my-sites/backup/main.jsx`, try that! 🙂

Reply:
@elliottprogrammer Ah! This worked great 💯  Thanks 👍